### PR TITLE
Fix ospfv2 module improperly rendering distance.admin_distance.

### LINKED ISSUE #1028: ios_ospfv2 module doesn't render admin_distance properly.
##### SUMMARY
The ospfv2 module doesn't properly render the distance.admin_distance section properly.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
ios_ospfv2

##### ANSIBLE VERSION

```paste below
ansible [core 2.15.8]
  config file = /home/taylorj/ansible/network/ansible.cfg
  configured module search path = ['/home/taylorj/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/taylorj/envs/ansible1/lib/python3.9/site-packages/ansible
  ansible collection location = /home/taylorj/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/taylorj/envs/ansible1/bin/ansible
  python version = 3.9.2 (default, Feb 28 2021, 17:03:44) [GCC 10.2.1 20210110] (/home/taylorj/envs/ansible1/bin/python3)
  jinja version = 3.1.2
  libyaml = True

```

##### COLLECTION VERSION

```paste below
Collection Version
---------- -------
cisco.ios  6.1.1
```

##### CONFIGURATION

```paste below
CONFIG_FILE() = /home/taylorj/ansible/network/ansible.cfg
DEFAULT_HOST_LIST(/home/taylorj/ansible/network/ansible.cfg) = ['/home/taylorj/ansible/network/hosts']
DEFAULT_VAULT_PASSWORD_FILE(/home/taylorj/ansible/network/ansible.cfg) = /home/taylorj/.ansible.pass
HOST_KEY_CHECKING(/home/taylorj/ansible/network/ansible.cfg) = False
```

##### OS / ENVIRONMENT

cisco 3650 Cisco IOS XE Software, Version 16.06.08

##### STEPS TO REPRODUCE
1. Create a playbook that has a cisco.ios.ios_ospfv2 with state rendered that includes an ospf process with admin distance.
2. 

```yaml
---
- name: setup interfaces
  hosts: all
  gather_facts: false
  connection: network_cli

  tasks:
    - cisco.ios.ios_ospfv2:
        config:
          processes:
            - process_id: 1
              distance:
                admin_distance:
                  distance: 1
        state: rendered

```

##### EXPECTED RESULTS

The rendered ios config should include the distance like this 
```
 "rendered": [
        "router ospf 1",
        "distance 1"
    ]
```
##### ACTUAL RESULTS

The config for the ospf process that is rendered doesn't include the admin distance.

```paste below

ok: [c49300a] => {
    "changed": false,
    "invocation": {
        "module_args": {
            "config": {
                "processes": [
                    {
                        "address_family": null,
                        "adjacency": null,
                        "areas": null,
                        "auto_cost": null,
                        "bfd": null,
                        "capability": null,
                        "compatible": null,
                        "default_information": null,
                        "default_metric": null,
                        "discard_route": null,
                        "distance": {
                            "admin_distance": {
                                "acl": null,
                                "address": null,
                                "distance": 1,
                                "wildcard_bits": null
                            },
                            "ospf": null
                        },
                        "distribute_list": null,
                        "domain_id": null,
                        "domain_tag": null,
                        "event_log": null,
                        "help": null,
                        "ignore": null,
                        "interface_id": null,
                        "ispf": null,
                        "limit": null,
                        "local_rib_criteria": null,
                        "log_adjacency_changes": null,
                       "max_lsa": null,
                        "max_metric": null,
                        "maximum_paths": null,
                        "mpls": null,
                        "neighbor": null,
                        "network": null,
                        "nsf": null,
                        "passive_interface": null,
                        "passive_interfaces": null,
                        "prefix_suppression": null,
                        "priority": null,
                        "process_id": 1,
                        "queue_depth": null,
                        "router_id": null,
                        "shutdown": null,
                        "summary_address": null,
                        "timers": null,
                        "traffic_share": null,
                        "ttl_security": null,
                        "vrf": null
                    }
                ]
            },
            "running_config": null,
            "state": "rendered"
        }
    },
    "rendered": [
        "router ospf 1"
    ]
}


```

The bug is in  the ospfv2  distance.admin_distance template where the setval doesn't properly qualify the properties.
```
diff --git a/plugins/module_utils/network/ios/rm_templates/ospfv2.py b/plugins/module_utils/network/ios/rm_templates/ospfv2.py
index 74b380c..8054825 100644
--- a/plugins/module_utils/network/ios/rm_templates/ospfv2.py
+++ b/plugins/module_utils/network/ios/rm_templates/ospfv2.py
@@ -721,9 +721,9 @@ class Ospfv2Template(NetworkTemplate):
                 $""",
                 re.VERBOSE,
             ),
-            "setval": "distance {{ admin_distance.distance }} "
-            "{{ ( admin_distance.address + ' ' + admin_distance.wildcard_bits ) if admin_distance.address is defined else '' }}"
-            "{{ ' ' + admin_distance.acl if admin_distance.acl is defined else '' }}",
+            "setval": "distance {{ distance.admin_distance.distance }} "
+            "{{ ( distance.admin_distance.address + ' ' + distance.admin_distance.wildcard_bits ) if distance.admin_distance.address is defined else '' }}"
+            "{{ ' ' + distance.admin_distance.acl if distance.admin_distance.acl is defined else '' }}",
             "result": {
                 "processes": {
                     "{{ pid }}": {
```

### PR DESCRIPTION
**Pull request by round3d**
##### SUMMARY

Fixes the ospfv2 distance.admin_distance template. 
Minor changes to qualify the properties in the setval of the template.

Fixes #1028 

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

ospfv2

##### ADDITIONAL INFORMATION

before:
```paste below

    "rendered": [
        "router ospf 1"
    ]
```
after:
```
    "rendered": [
        "router ospf 1",
        "distance 1 "
    ]


```

### CODE REVIEW

**Comment by KB-perByte** [comment]
Hey @round3d Thank you for your contribution, can you please add a changelog and a unit test case to confirm your changes?
Let us know if you need any help.
Regards

**Comment by round3d** [comment]
Hi @KB-perByte , 
I can do that with the changelog.

The existing test case test_ios_ospfv2_rendered (and some others that include distance.admin_distance) should already cover it, but the commands list for the assert doesn't include the admin lines that should be rendered.  It includes the distance ospf config but not the distance for admin_distance. That's the cause of some of those checks failing.


https://github.com/ansible-collections/cisco.ios/blob/cefdf849fb395ebca2b49b342a524fa9e4430680/tests/unit/modules/network/ios/test_ios_ospfv2.py#L753-L765

https://github.com/ansible-collections/cisco.ios/blob/cefdf849fb395ebca2b49b342a524fa9e4430680/tests/unit/modules/network/ios/test_ios_ospfv2.py#L915-L950

Sorry if that's not clear.

**Review by roverflow — CHANGES_REQUESTED** [review]
Please fix the tests and add a changelog as mentioned above

**Comment by round3d** [comment]
> Please fix the tests and add a changelog as mentioned above

Hi @roverflow , I've done that.  Thanks.

File: changelogs/fragments/ospfv2_admin_distance_fix.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+bugfixes:
+  - ospfv2 - Fixed rendering of admin_distance.
```
Comment (roverflow) [inline changelogs/fragments/ospfv2_admin_distance_fix.yml:3]:
```suggestion
  - ios_ospfv2 - Fix improper rendering of admin_distance attribute.
```

**Review by roverflow — APPROVED** [review]

**Pull request merged**